distributed/core.py

```diff
@@ -229,11 +229,36 @@ def write(stream, msg):
     stream.write(b''.join(lengths))
 
     for frame in frames:
+        # Can't wait for the write() Future as it may be lost
+        # ("If write is called again before that Future has resolved,
+        #   the previous future will be orphaned and will never resolve")
         stream.write(frame)
 
     yield gen.moment
 
 
+@gen.coroutine
+def flush(stream):
+    """Flush the stream's output buffer.
+    This is recommended before closing the stream.
+    """
+    if stream.writing():
+        yield stream.write(b'')
```

Comment:
I'm mostly concerned about us as users.  Only very advanced dask/distributed users should use read/write/close directly.  

However, given that we've had problems reported it's possible that we aren't handling everything well internally.  

I have no problem with read/write/close as an API generally.